zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala

```diff
@@ -11,8 +11,12 @@ import zhttp.http.HeaderValues
  * true or false based on if the conditions are met or not.
  */
 trait HeaderChecks[+A] { self: HeaderExtension[A] with A =>
-  final def hasContentType(value: CharSequence): Boolean =
-    contentType.exists(contentEqualsIgnoreCase(value, _))
+  final def hasContentType(value: CharSequence): Boolean = {
+    contentType.exists(h => {
+      val max = Math.min(value.length, h.length)
+      contentEqualsIgnoreCase(h.subSequence(0, max), value)
+    })
+  }
```

Comment:
We can use `contentStartsWithIgnoreCase` which will be more performant.

Reply:
Can one not use `mediaType` field on Headers? Something like -

```scala
headers.mediaType match {
  case Some(MediaType.application.json) => true
  case None                             => false
}
```
We can wrap this into a function `hasMediaType` and add it to HeaderChecks also.

Reply:
> We can use contentStartsWithIgnoreCase which will be more performant.

I can't seem to find that method, but I can absolutely implement it if we want to!

> Can one not use mediaType field on Headers? Something like -

Not right now, because `mediaType` suffers from the same problem:

```
  final def mediaType: Option[MediaType] =
    contentType.flatMap(ct => MediaType.forContentType(ct.toString))
```

Reply:
There is a `startsWith` in `AsciiString` may be we can use that?

Reply:
> There is a startsWith in AsciiString may be we can use that?

It will be used under the hood by `contentEqualsIgnoreCase` is `a` or `b` is an AsciiString 👍